### PR TITLE
Add APIs to supply a vclock when doing a delete

### PR DESCRIPTION
**Pull request by Vagabond**

### CODE REVIEW

**Comment by jonmeredith** [comment]
The eunit tests fail for me now

*\* Found 132 name clashes in code paths 
  riak_kv_delete: delete_test_..._timed out_
  undefined

**Comment by Vagabond** [comment]
Eunit tests fixed, RW is expanded as before.

File: src/riak_kv_delete.erl [hunk 1]
```diff
@@ -28,52 +28,75 @@
...
+    case get_rw_val(Bucket, RW0) of
         error ->
             Client ! {ReqId, {error, {rw_val_violation, RW0}}};
         RW ->
-            {ok,C} = riak:local_client(ClientId),
+            RealStartTime = riak_core_util:moment(),
+            {ok, C} = riak:local_client(),
             case C:get(Bucket,Key,RW,Timeout) of
                 {ok, OrigObj} ->
+                    %?debugFmt("doing a delete of ~p~n", [OrigObj]),
```
Comment (jonmeredith) [inline src/riak_kv_delete.erl]: debug cruft to remove

File: src/riak_kv_delete.erl [hunk 2]
```diff
@@ -28,52 +28,75 @@
...
-                    Client ! {ReqId, Reply},
-                    case Reply of
-                        ok ->
-                            {ok, C2} = riak:local_client(),
-                            C2:get(Bucket, Key, N, RemainingTime);
-                        _ -> nop
-                    end;
+                    delete(ReqId,Bucket,Key,RW,RemainingTime,Client,ClientId,riak_object:vclock(OrigObj));
                 {error, notfound} ->
+                    %?debugFmt("notfound ~n", []),
```
Comment (jonmeredith) [inline src/riak_kv_delete.erl]: more debug

File: src/riak_kv_delete.erl [hunk 3]
```diff
@@ -28,52 +28,75 @@
...
                     Client ! {ReqId, {error, notfound}};
                 X ->
+                    %?debugFmt("error ~p ~n", [X]),
```
Comment (jonmeredith) [inline src/riak_kv_delete.erl]: more debug

**Comment by jonmeredith** [comment]
Remove the commented out debugging code and it's go for merge.
Unit tests pass now.

**Comment by Vagabond** [comment]
Merged

**Pull request closed without merging**